.SolarisJenkinsfile

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env groovy
+
+pipeline {
+  agent {
+    label 'solaris'
+  }
+
+  environment {
+    PATH = "/opt/local/bin:$PATH"
+    CC = 'gcc'
+    CXX = 'g++'
+  }
+
+  options {
+      timestamps()
+  }
+
+  stages {
+    stage('GitHub Checkout') {
+      steps {
+        checkout scm
+      }
+    }
+
+    stage('Build') {
+      steps {
+        sh "mkdir -p build"
+
+        dir("build") {
+          sh "cmake -DCMAKE_BUILD_TYPE=Debug -DST_BUILD_TESTS=ON -DCMAKE_INSTALL_PREFIX=${env.WORKSPACE}/artifacts .."
+          sh "make"
+        }
+      }
+    }
+
+    stage('Test') {
+      steps {
+        dir("build") {
+          sh "make test"
+        }
+      }
+    }
+
+    stage('Install') {
+      steps {
+        dir("build") {
+          sh "make install"
+        }
+      }
+
+      post {
+        success {
+          dir("artifacts") {
+            archiveArtifacts artifacts: '**/*', fingerprint: true, onlyIfSuccessful: true
+          }
+        }
+      }
+    }
+  }
+
+  post {
+    always {
+      emailext attachLog: true,
+               body: '''${JELLY_SCRIPT, template="html"}''',
+               recipientProviders: [developers(), requestor()],
```

Comment:
Who do these expand to?

Reply:
`requestor()` is the email address of the Jenkins user (if any) who manually triggered a build.

`developers()` should be the emails of the commit authors for the range of commits since the previous build. I'm not sure whether this requires there to be a matching Jenkins user with that email address or not.

Reply:
Is there a way to have this report back build status to the commit in GitHub, as we do with GH Actions and AppVeyor?

Reply:
I don't believe that is possible without one of us generating a Personal Access Token and making it available in Jenkins :disappointed:

Reply:
If my GitHub account can post that notification to your repository or build hook, I can produce the token and hook it in.